src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -172,18 +172,27 @@
   <data name="AppDomain_Policy_PrincipalTwice" xml:space="preserve">
     <value>Default principal object cannot be set twice.</value>
   </data>
-  <data name="AmbiguousImplementationException_NullMessage" xml:space="preserve">
-    <value>Ambiguous implementation found.</value>
-  </data>
   <data name="Arg_AccessException" xml:space="preserve">
     <value>Cannot access member.</value>
   </data>
   <data name="Arg_AccessViolationException" xml:space="preserve">
     <value>Attempted to read or write protected memory. This is often an indication that other memory is corrupt.</value>
   </data>
-  <data name="Arg_AmbiguousMatchException" xml:space="preserve">
+  <data name="Arg_AmbiguousImplementationException_NoMessage" xml:space="preserve">
+    <value>Ambiguous implementation found.</value>
+  </data>
+  <data name="Arg_AmbiguousMatchException_Attribute" xml:space="preserve">
+    <value>Multiple custom attributes of the same type '{0}' found.</value>
+  </data>
+  <data name="Arg_AmbiguousMatchException_NoMessage" xml:space="preserve">
     <value>Ambiguous match found.</value>
   </data>
+  <data name="Arg_AmbiguousMatchException_Simple" xml:space="preserve">
+    <value>Ambiguous match for '{0}' found.</value>
+  </data>
+  <data name="Arg_AmbiguousMatchException_Detailed" xml:space="preserve">
+    <value>Ambiguous match for '{0} {1}' found.</value>
```

Comment:
I am still not sure we really need this version with `memberInfo.DeclaringType` added, for me just `Ambiguous match for '{0}' found.` with the only `memberInfo` passed is good enough.

Reply:
That's mostly helpful for the `AttributeData` case, but I can go either way :)